test/specs/scale.time.tests.js

```diff
@@ -584,6 +584,119 @@ describe('Time scale tests', function() {
 		expect(xScale.getLabelForIndex(6, 0)).toBe('2015-01-10T12:00');
 	});
 
+	it('should get the correct label when time is specified as a string', function() {
+		var chart = window.acquireChart({
+			type: 'line',
+			data: {
+				datasets: [{
+					xAxisID: 'xScale0',
+					data: [{t: '2015-01-01T20:00:00', y: 10}, {t: '2015-01-02T21:00:00', y: 3}]
+				}],
+			},
+			options: {
+				scales: {
+					xAxes: [{
+						id: 'xScale0',
+						type: 'time',
+						position: 'bottom'
+					}],
+				}
+			}
+		});
+
+		var xScale = chart.scales.xScale0;
+		expect(xScale.getLabelForIndex(0, 0)).toBeTruthy();
+		expect(xScale.getLabelForIndex(0, 0)).toBe('2015-01-01T20:00:00');
+	});
+
+	it('should get the correct label for a timestamp with milliseconds', function() {
+		var chart = window.acquireChart({
+			type: 'line',
+			data: {
+				datasets: [{
+					xAxisID: 'xScale0',
+					data: [{t: 1515469457180, y: 10}, {t: 1515469458180, y: 3}]
+				}],
+			},
+			options: {
+				scales: {
+					xAxes: [{
+						id: 'xScale0',
+						type: 'time',
+						position: 'bottom'
+					}],
+				}
+			}
+		});
+
+		var xScale = chart.scales.xScale0;
+		var label0 = xScale.getLabelForIndex(0, 0);
+		// check things that won't differ by timezone until we have timezone support
+		expect(label0).toBeTruthy();
+		expect(label0).toContain('Jan');
+		expect(label0).toContain('2018');
+		expect(label0).toContain('44:17.180');
+		expect(label0).toContain('m');
+	});
+
+	it('should get the correct label for a timestamp with time', function() {
+		var chart = window.acquireChart({
+			type: 'line',
+			data: {
+				datasets: [{
+					xAxisID: 'xScale0',
+					data: [{t: 1515469457000, y: 10}, {t: 1515469458000, y: 3}]
+				}],
+			},
+			options: {
+				scales: {
+					xAxes: [{
+						id: 'xScale0',
+						type: 'time',
+						position: 'bottom'
+					}],
+				}
+			}
+		});
+
+		var xScale = chart.scales.xScale0;
+		var label0 = xScale.getLabelForIndex(0, 0);
+		// check things that won't differ by timezone until we have timezone support
+		expect(label0).toBeTruthy();
+		expect(label0).toContain('Jan');
+		expect(label0).toContain('2018');
+		expect(label0).not.toContain('.000');
```

Comment:
This test and the following matches both `'MMM D, YYYY h:mm:ss a'` and `'MMM D, YYYY'` so will not fail if the result change. I think it's better to use regex for all tests of that PR, something like (not tested):

```javascript
expect(label).toMatch(/^Jan \d+, 2018 \d{1,2}:\d{2}:\d{2}.\d{3} (am|pm)$/); // millisecond
expect(label).toMatch(/^Jan \d+, 2018 \d{1,2}:\d{2}:\d{2} (am|pm)$/); // time
expect(label).toMatch(/^Jan \d+, 2018$/); // date
```

Reply:
done

Reply:
@simonbrunel the problem with that is that it doesn't work across timezones. Midnight in one timezone gives the date formatting, but is 1am in the next timezone and will give the time formatting. All of our tests are pretty flaky in this way. We have lots of tests that should check if `date == a` and instead check if `date == a || date == b` for this reason. I'd like to implement timezone support, but it will have to be a separate issue saved for another day

Reply:
True, that's really annoying: can't we simply generate the timestamp from the local timezone and thus ensure that we get the expected date/time, maybe something like that:

```javascript
data: [
  {t: +new Date('2018-01-08 00:00:00'), y: 10},
  {t: +new Date('2018-01-09 00:00:00'), y: 3}
]
```

If that works, we don't even need the regex :)

Reply:
Good idea. I updated it that way. Much better solution